Formula/nu.rb

```diff
@@ -15,9 +15,44 @@ class Nu < Formula
 
   depends_on "pcre"
 
+  uses_from_macos "llvm" => [:build, :test]
+  uses_from_macos "swift" => :build # For libdispatch on Linux.
+  uses_from_macos "libffi"
+
+  on_linux do
+    depends_on "gnustep-make" => :build
+    depends_on "gnustep-base"
+    depends_on "libobjc2"
+  end
+
+  # Clang must be used on Linux because GCC Objective-C support is insufficient.
+  fails_with :gcc
+
   def install
     ENV.delete("SDKROOT") if MacOS.version < :sierra
     ENV["PREFIX"] = prefix
+    # Don't hard code path to clang.
+    inreplace "tools/nuke", "/usr/bin/clang", ENV.cc
+
+    unless OS.mac?
+      ENV.append_path "PATH", Formula["gnustep-make"].libexec
+
+      # Help linker find libdispatch from swift on Linux.
+      # This is only used for the mininush temporary compiler and is not needed for nush.
+      ldflags = %W[
+        "-L#{Formula["swift"].libexec}/lib/swift/linux"
+        "-Wl,-rpath,#{Formula["swift"].libexec}/lib/swift/linux"
+      ]
+      ENV["LIBDIRS"] = ldflags.join(" ")
+
+      # Remove unused prefix from ffi.h
+      inreplace ["objc/NuBridge.h", "objc/NuBridge.m", "objc/Nu.m"], "x86_64-linux-gnu/", ""
+
+      # Remove CFLAGS that force using GNU runtime on Linux.
+      # Remove this workaround when upstream drops these flags or provides a way to disable them.
```

Comment:
Has it been reported upstream?

Reply:
I will open an issue on their GitHub asking if they can look into making this optional on Linux.

Reply:
I've updated this to include a link to the GitHub issue I opened.